src/utils/resolveVariables.ts

```diff
@@ -4,44 +4,78 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as path from 'path';
-import { window, WorkspaceFolder } from 'vscode';
+import { window, workspace, WorkspaceFolder } from 'vscode';
 import { cloneObject } from '../utils/cloneObject';
 
-export function resolveVariables<T>(target: T, folder: WorkspaceFolder): T {
+const variableMatcher: RegExp = /\$\{[a-z\.\-_:]+\}/ig;
+const configVariableMatcher: RegExp = /\$\{config:([a-z\.\-_]+)\}/i;

```

Comment:
No `g` flag?

Reply:
For this, yes. `variableMatcher` is used to find all instances of anything that looks like `${something}`, including `${config:something.something}`, within the entire input string. `configVariableMatcher` is used only to match against a specific single variable--so there's nothing before or after it--and also, to separate the `something.something` config parameter into a regex group. `g` isn't needed here since we aren't looking for multiple `${config:something.something}` entries within the string.